aws_lambda_powertools/event_handler/middlewares/openapi_validation.py

```diff
@@ -383,15 +386,18 @@ def _normalize_multi_query_string_with_param(query_string: Optional[Dict[str, st
     -------
     A dictionary containing the processed multi_query_string_parameters.
     """
-    if query_string:
+    if not query_string:
+        return None
+    else:
+        resolved_query_string: Dict[str, Any] = query_string
         for param in filter(is_scalar_field, params):
             try:
                 # if the target parameter is a scalar, we keep the first value of the query string
                 # regardless if there are more in the payload
-                query_string[param.alias] = query_string[param.alias][0]
+                resolved_query_string[param.alias] = query_string[param.alias][0]
             except KeyError:
                 pass
-    return query_string
+        return resolved_query_string
 
```

Comment:
I've changed these lines to make typing work, but this is not how the bug was fixed. The bug was fixed by making sure all event handler types return `Dict[str, List[str]]` when resolving query string parameters.